File: 0-999/600-699/660-669/665/generate_testcases.go
package main

import (
	"fmt"
	"math/rand"
	"os"
)

func genA() {
	rand.Seed(1)
	f, _ := os.Create("testcasesA.txt")
	defer f.Close()
	const dayStart = 5 * 60
	const dayEnd = 23*60 + 59
	t := 100
	fmt.Fprintln(f, t)
	for i := 0; i < t; i++ {
		a := rand.Intn(120) + 1
		ta := rand.Intn(120) + 1
		b := rand.Intn(120) + 1
		tb := rand.Intn(120) + 1
		maxTrips := (dayEnd-dayStart)/a + 1
		start := dayStart + rand.Intn(maxTrips)*a
		hh := start / 60
		mm := start % 60
		fmt.Fprintf(f, "%d %d %d %d %d %d\n", a, ta, b, tb, hh, mm)
	}
}

func randPerm(n int) []int {
	arr := make([]int, n)
	for i := 0; i < n; i++ {
		arr[i] = i + 1
	}
	rand.Shuffle(n, func(i, j int) { arr[i], arr[j] = arr[j], arr[i] })
	return arr
}

func pickDistinct(k, m int) []int {
	arr := randPerm(k)
	return arr[:m]
}

func genB() {
	rand.Seed(2)
	f, _ := os.Create("testcasesB.txt")
	defer f.Close()
	t := 100
	fmt.Fprintln(f, t)
	for i := 0; i < t; i++ {
		k := rand.Intn(8) + 1
		m := rand.Intn(k) + 1
		n := rand.Intn(5) + 1
		row := randPerm(k)
		fmt.Fprintf(f, "%d %d %d ", n, m, k)
		for j := 0; j < k; j++ {
			fmt.Fprintf(f, "%d ", row[j])
		}
		for c := 0; c < n; c++ {
			order := pickDistinct(k, m)
			for j := 0; j < m; j++ {
				fmt.Fprintf(f, "%d ", order[j])
			}
		}
		fmt.Fprintln(f)
	}
}

func randString(n int) string {
	b := make([]byte, n)
	for i := range b {
		b[i] = byte('a' + rand.Intn(26))
	}
	return string(b)
}

func genC() {
	rand.Seed(3)
	f, _ := os.Create("testcasesC.txt")
	defer f.Close()
	t := 100
	fmt.Fprintln(f, t)
	for i := 0; i < t; i++ {
		n := rand.Intn(50) + 1
		s := randString(n)
		fmt.Fprintln(f, s)
	}
}

func genD() {
	rand.Seed(4)
	f, _ := os.Create("testcasesD.txt")
	defer f.Close()
	t := 100
	fmt.Fprintln(f, t)
	for i := 0; i < t; i++ {
		n := rand.Intn(30) + 1
		fmt.Fprintf(f, "%d ", n)
		for j := 0; j < n; j++ {
			v := rand.Intn(100) + 1
			fmt.Fprintf(f, "%d ", v)
		}
		fmt.Fprintln(f)
	}
}

func genE() {
	rand.Seed(5)
	f, _ := os.Create("testcasesE.txt")
	defer f.Close()
	t := 100
	fmt.Fprintln(f, t)
	for i := 0; i < t; i++ {
		n := rand.Intn(50) + 1
		k := rand.Intn(1000) + 1
		fmt.Fprintf(f, "%d %d ", n, k)
		for j := 0; j < n; j++ {
			v := rand.Intn(1000)
			fmt.Fprintf(f, "%d ", v)
		}
		fmt.Fprintln(f)
	}
}

func genF() {
	rand.Seed(6)
	f, _ := os.Create("testcasesF.txt")
	defer f.Close()
	t := 100
	fmt.Fprintln(f, t)
	for i := 0; i < t; i++ {
		n := rand.Int63n(1000000) + 1
		fmt.Fprintln(f, n)
	}
}

func main() {
	genA()
	genB()
	genC()
	genD()
	genE()
	genF()
}

File: 0-999/900-999/980-989/981/generate_testcases.py
import random, string
random.seed(0)

def gen_A(n=100):
    with open('testcasesA.txt','w') as f:
        for _ in range(n):
            l=random.randint(1,50)
            s=''.join(random.choice(string.ascii_lowercase) for _ in range(l))
            f.write(s+'\n')

def gen_B(n=100):
    with open('testcasesB.txt','w') as f:
        for _ in range(n):
            n1=random.randint(1,5)
            f.write(str(n1)+'\n')
            indices=random.sample(range(1,30), n1)
            for a in indices:
                x=random.randint(1,100)
                f.write(f"{a} {x}\n")
            m=random.randint(1,5)
            f.write(str(m)+'\n')
            indices2=random.sample(range(1,30), m)
            for b in indices2:
                y=random.randint(1,100)
                f.write(f"{b} {y}\n")

def gen_C(n=100):
    with open('testcasesC.txt','w') as f:
        for _ in range(n):
            nodes=random.randint(2,8)
            f.write(str(nodes)+'\n')
            for i in range(2,nodes+1):
                p=random.randint(1,i-1)
                f.write(f"{p} {i}\n")

def gen_D(n=100):
    with open('testcasesD.txt','w') as f:
        for _ in range(n):
            n1=random.randint(1,8)
            k=random.randint(1,n1)
            f.write(f"{n1} {k}\n")
            arr=[random.randint(0,1000) for _ in range(n1)]
            f.write(' '.join(map(str,arr))+'\n')

def gen_E(n=100):
    with open('testcasesE.txt','w') as f:
        for _ in range(n):
            n1=random.randint(1,8)
            q=random.randint(1,5)
            f.write(f"{n1} {q}\n")
            for _ in range(q):
                l=random.randint(1,n1)
                r=random.randint(l,n1)
                x=random.randint(1,n1)
                f.write(f"{l} {r} {x}\n")

def gen_F(n=100):
    with open('testcasesF.txt','w') as f:
        for _ in range(n):
            n1=random.randint(1,5)
            L=random.randint(10,100)
            f.write(f"{n1} {L}\n")
            a=[random.randint(0,L) for _ in range(n1)]
            b=[random.randint(0,L) for _ in range(n1)]
            f.write(' '.join(map(str,a))+'\n')
            f.write(' '.join(map(str,b))+'\n')

def gen_G(n=100):
    with open('testcasesG.txt','w') as f:
        for _ in range(n):
            n1=random.randint(1,4)
            q=random.randint(1,5)
            f.write(f"{n1} {q}\n")
            for _ in range(q):
                t=random.randint(1,2)
                if t==1:
                    l=random.randint(1,n1)
                    r=random.randint(l,n1)
                    x=random.randint(1,n1)
                    f.write(f"1 {l} {r} {x}\n")
                else:
                    l=random.randint(1,n1)
                    r=random.randint(l,n1)
                    f.write(f"2 {l} {r}\n")

def gen_H(n=100):
    with open('testcasesH.txt','w') as f:
        for _ in range(n):
            nodes=random.randint(2,5)
            k=random.randint(1,min(3,nodes))
            f.write(f"{nodes} {k}\n")
            for i in range(2,nodes+1):
                p=random.randint(1,i-1)
                f.write(f"{p} {i}\n")

def main():
    gen_A()
    gen_B()
    gen_C()
    gen_D()
    gen_E()
    gen_F()
    gen_G()
    gen_H()

if __name__=='__main__':
    main()
